### ISSUE TITLE
Job class missing methods in Qiskit spec

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-ibmq-provider#491]
### What is the expected behavior?
The qiskit spec has the following for the jobs class:

>The job also has additional calls Job.done(), Job.running() and Job.cancelled() which returns a boolean (true/false) to indicate whether the job is done, running, or cancelled.

Do we want these?  In my experience this would be super nice because checking if jobs are done is currently confusing / ugly.

**Comment by ewinston** [comment]
This is not a comment about whether we want these but if we did it would probably be better to put in terra basejob for consistency. I liked having them but I believe they placed a burden on the servers since they are often used in loops. Not sure if recent updates to the api changes things?

**Comment by nonhermitian** [comment]
Not sure how this is different than people just calling `job.status()` repeatedly in loops.  I think the websockets take care of this anyway.  The nice thing with these is that it is simply to do `if job.done()`.  Currently one has to know that `if job.status() == job.status().DONE` is the way to do it, but this is really not intuitive.  Similarly with using the`JobStatus` itself, which is not imported by default as well.

**Comment by jyu00** [comment]
> I liked having them but I believe they placed a burden on the servers since they are often used in loops.

That is indeed an issue. The api's solution is to use websocket, but the provider's implementation today doesn't update the user with intermediate status. There is an [issue opened in Aqua](https://github.com/Qiskit/qiskit-aqua/issues/752) on this, where they call `job.status()` (which doesn't use websocket) in a loop. The provider will hopefully address this in the near future.

**Comment by nonhermitian** [comment]
So the changes presented here do not change the status quo, that is to say an end user can query the status as much as they like and get blocked by the IBM cloud.  So I do not understand why having convenience functions that make checking the status easier are an issue? The Aqua issue is just developers not understanding the limitations of this call.  The monitoring tools in Terra know to adaptively change the status check interval based on job position.

**Comment by jyu00** [comment]
I agree these convenience functions are nice to have. I was just pointing out that calling `job.status()` in a loop is a known issue that will be addressed separately.

**Comment by ewinston** [comment]
A local simulator doesn't mind putting this in a loop so it could be in basejob but overridden by providers as they see fit.

**Comment by jyu00** [comment]
Since it's in the spec it seems reasonable to have them in `BaseJob`.

**Comment by jyu00** [comment]
@luisg5 since the implementation doesn't have anything that's ibmq-provider specific, can you move it to `BaseJob` (in terra)? Then all providers can benefit from them implicitly.

**Comment by jyu00** [comment]
Remove these methods from the provider now that they're in terra (once terra is released).
